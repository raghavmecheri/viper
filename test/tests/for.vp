int j;
for (int i = 0; i < 5; i+=1) {
    print("Outer:");
    print(i);
    for (j = i; j < 10; j+=1) {
        print("Inner:");
        print(j);
    }
}

/*
for(int f in arr) {
    j = j + f;
}

int g;

for(g in arr) {
    j = j + g;
}

for((int, bool) tup in list) {
    if (tup[1] == true) {
        j += tup[0];
    }
}

for((int iter, int index) in list) {
    j += index;
    j += iter;
}
*/
