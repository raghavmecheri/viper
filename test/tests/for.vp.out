Outer:
0
Inner:
0
Inner:
1
Inner:
2
Inner:
3
Inner:
4
Inner:
5
Inner:
6
Inner:
7
Inner:
8
Inner:
9
Outer:
1
Inner:
1
Inner:
2
Inner:
3
Inner:
4
Inner:
5
Inner:
6
Inner:
7
Inner:
8
Inner:
9
Outer:
2
Inner:
2
Inner:
3
Inner:
4
Inner:
5
Inner:
6
Inner:
7
Inner:
8
Inner:
9
Outer:
3
Inner:
3
Inner:
4
Inner:
5
Inner:
6
Inner:
7
Inner:
8
Inner:
9
Outer:
4
Inner:
4
Inner:
5
Inner:
6
Inner:
7
Inner:
8
Inner:
9